arangod/GeneralServer/CommTask.cpp

```diff
@@ -150,6 +153,11 @@ bool resolveRequestContext(GeneralRequest& req) {
 
 CommTask::Flow CommTask::prepareExecution(auth::TokenCache::Entry const& authToken,
                                           GeneralRequest& req) {
+
+#ifdef USE_DTRACE
```

Comment:
why  DTRACE_PROBE1 not just a noop, instead of wrapping everything in #ifdef

Reply:
There are 74 #define statements in `lib/Basics/sdt.h`. Yes, so far I am using only two of them. But I think it is impractical to define them all as empty macros if `USE_DTRACE` is set to 0. I would prefer to leave it as it is, but it is an easy change to define those that we actually use as empty macros for now, if you strongly prefer this. I have no strong feelings.

Reply:
I would also prefer having just one file to be included wherever tracing is needed:
lib/Basics/dtrace-wrapper.h

And this file could contain:
```
#ifndef ARANGODB_DTRACE_WRAPPER_H
#define ARANGODB_DTRACE_WRAPPER_H 1

#if USE_DTRACE
#include "Basics/sdt.h"
#else 
#define DTRACE_PROBE1(...) do { } while (0);
#define DTRACE_PROBE2(...) do { } while (0);
#endif 

#endif 
```
That would allow us to have everywhere managed in a single place, and keep the changes to the code minimal.

Reply:
I like this approach.